### PR TITLE
Laget metode for aa kun returnere norske arbeidsforhold i perioden

### PR DESCRIPTION
**Pull request by siljerisvik1**
Kan noe slikt funke for å sjekke landekode til arbeidsgiver i arbeidsforholdene alltid er norsk innenfor perioden? Gjerne kom med tilbakemeldinger for forbredringer i koden.

### CODE REVIEW

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 1]
```diff
@@ -19,7 +19,20 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
 
     fun arbeidsforhold(): List<Arbeidsforhold> = datagrunnlag.arbeidsforhold
 
-    fun sisteArbeidsgiversLand(): String? = datagrunnlag.arbeidsforhold[0].arbeidsgiver.landkode
+    fun sisteArbeidsgiversLand(): String? {
+
+        val arbeidsforholdIPeriode = datagrunnlag.arbeidsforhold.filter {
+            it.periode.fom!! >= datagrunnlag.periode.fom && it.periode.tom!! <= datagrunnlag.periode.tom}
```
Comment (haagenhasle) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: Er fom egentlig relevant? Hvis arbeidsforholdet startet før perioden vi ser på, så er det jo like fullt gyldig i perioden vi ser på? Og hvis det startet etter perioden vi ser på, så er det også relevant..?

Comment (siljerisvik1) [reply src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: Ja, stemmer nok det. Må tenke litt mer på den.

File: src/main/kotlin/no/nav/medlemskap/domene/Arbeidsforhold.kt [hunk 2]
```diff
@@ -43,6 +43,13 @@ enum class Skipsregister (val beskrivelse: String) {
     utl("Utenlandsk skipsregister")
 }
 
+enum class LuftfartYrkeskoder(val beskrivelse: String){
+    PILOT("13143107"),
+    KABINPERSONALE(	"5111105"),
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/domene/Arbeidsforhold.kt]: Space?

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 3]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
 
     fun personensDokumenterIJoark(): List<Journalpost> = datagrunnlag.dokument
 
-    fun personensSisteStatsborgerskap(): String = datagrunnlag.personhistorikk.statsborgerskap[0].landkode
+    fun personensSisteStatsborgerskap(): String {
+        val periodeDatagrunnlag = Interval.of(datagrunnlag.periode.fom.atStartOfDay(ZoneId.systemDefault()).toInstant(), datagrunnlag.periode.tom.atStartOfDay(ZoneId.systemDefault()).toInstant())
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: Denne kan byttes ut med lagIntervall

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 4]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
...
+        val statsborgerskapIPeriode = datagrunnlag.personhistorikk.statsborgerskap.filter {
+            periodeDatagrunnlag.overlaps(lagInterval(Periode(it.fom, it.tom))) ||
+            periodeDatagrunnlag.encloses(lagInterval(Periode(it.fom, it.tom)))
+        }
+
+        statsborgerskapIPeriode.forEach{
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: return statsborgerskapIPeriode.find { !eøsLand.containsKey(it.landkode) } ?: statsborgerskapIPeriode.last().landkode

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 5]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
...
+
 
     fun arbeidsforhold(): List<Arbeidsforhold> = datagrunnlag.arbeidsforhold
 
-    fun sisteArbeidsgiversLand(): String? = datagrunnlag.arbeidsforhold[0].arbeidsgiver.landkode
+    fun sisteArbeidsgiversLand(): String? {
+
+        val arbeidsforholdPeriode = hentArbeidsforholdIPeriode()
+
+        arbeidsforholdPeriode.forEach{
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: du kan bruke `find` her også

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 6]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
...
+    }
+
+    private fun lagInterval(periode: Periode): Interval {
+        val fom = periode.fom ?: LocalDate.MIN
+        val tom = periode.tom ?: LocalDate.MAX
+        return Interval.of(fom.atStartOfDay(ZoneId.systemDefault()).toInstant(), tom.atStartOfDay(ZoneId.systemDefault()).toInstant())
+    }
+
+    fun sisteArbeidsforholdtype(): Arbeidsforholdstype {
+        val arbeidsforholdPeriode = hentArbeidsforholdIPeriode()
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: `find` her også

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 7]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
...
 
-    fun sisteArbeidsforholdSkipsregister(): Skipsregister? = datagrunnlag.arbeidsforhold[0].arbeidsavtaler[0].skipsregister
+    fun sisteArbeidsforholdYrkeskode(): String {
+        val yrkeskoderLuftfart = listOf(LuftfartYrkeskoder.KABINPERSONALE.beskrivelse,
+                                        LuftfartYrkeskoder.KABINSJEF.beskrivelse,
+                                        LuftfartYrkeskoder.PILOT.beskrivelse)
+
+        val arbeidsforholdPeriode = hentArbeidsforholdIPeriode()
+
+        arbeidsforholdPeriode.forEach{ it ->
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: `flatMap` + `find` Spør meg gjerne :)

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 8]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
...
+
+    fun sisteArbeidsforholdSkipsregister(): Skipsregister? {
+
+        val periodeDatagrunnlag = Interval.of(datagrunnlag.periode.fom.atStartOfDay(ZoneId.systemDefault()).toInstant(), datagrunnlag.periode.tom.atStartOfDay(ZoneId.systemDefault()).toInstant())
+        val arbeidsforholdPeriode = datagrunnlag.arbeidsforhold.filter {
+            lagInterval(Periode(it.periode.tom, it.periode.fom)).overlaps(periodeDatagrunnlag) &&
+            lagInterval(Periode(it.periode.tom, it.periode.fom)).encloses(periodeDatagrunnlag)
+        }
+
+        arbeidsforholdPeriode.forEach{ it ->
```
Comment (richardborge) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: `flatMap` + `find` Spør meg gjerne :)

File: src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt [hunk 9]
```diff
@@ -15,17 +19,102 @@ class Personfakta(private val datagrunnlag: Datagrunnlag) {
...
+    private fun lagInterval(periode: Periode): Interval {
+        val fom = periode.fom ?: LocalDate.MIN
+        val tom = periode.tom ?: LocalDate.MAX
+        return Interval.of(fom.atStartOfDay(ZoneId.systemDefault()).toInstant(), tom.atStartOfDay(ZoneId.systemDefault()).toInstant())
+    }
+
+    fun sisteArbeidsforholdtype(): Arbeidsforholdstype {
+        val arbeidsforholdPeriode = hentArbeidsforholdIPeriode()
+        arbeidsforholdPeriode.forEach{
+            if(!it.arbeidsgiver.landkode.equals(Arbeidsforholdstype.NORMALT.toString())){
```
Comment (haagenhasle) [inline src/main/kotlin/no/nav/medlemskap/regler/common/Personfakta.kt]: Her ser det ut som om du sammenligner en landkode med en arbeidsforholdtype?

File: src/test/kotlin/no/nav/medlemskap/regler/v1/RegelsettForGrunnforordningenTest.kt [hunk 10]
```diff
@@ -16,6 +16,16 @@ class RegelsettForGrunnforordningenTest {
         assertEquals(Resultattype.JA, evaluer(personleser.enkelNorsk()))
     }
 
+    @Test
+    fun `person med flere statsborgerskap i periode får nei`() {
+        assertEquals(Resultattype.NEI, evaluer(personleser.norskMedFlereStatsborgerskap()))
+    }
+
+    @Test
+    fun `person med flere statsborgerskap som overlapper periode får nei`() {
```
Comment (haagenhasle) [inline src/test/kotlin/no/nav/medlemskap/regler/v1/RegelsettForGrunnforordningenTest.kt]: Flott at du legger til tester! Vurder gjerne om du kan legge til enda flere, for å sjekke ulike edge cases. Dekker vi "periode inni annen periode", "periode rundt annen periode", "periode som overlapper annen periode", osv.. ?
